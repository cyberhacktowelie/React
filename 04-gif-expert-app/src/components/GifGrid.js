// import React, { useState, useEffect } from 'react';
import { PropTypes } from 'prop-types';

import { GifGridItem } from './GifGridItem';
// import { getGifs } from '../helpers/getGifs';
import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {

    // const [ images, setImages ] = useState([]);

    const { data:images , loading } = useFetchGifs(category);

    // useEffect(() => {
    //     // getGifs();
    //     // getGifs(category).then(img => setImages(img));  
    //     getGifs(category).then(setImages);
    // }, [ category ]);


    // const apiKey = 'QKXFCiGkEFDCSb1fyIPKGNNZ2wIx1E5g';
    // const getGifs = async () => {
    //     const url = `https://api.giphy.com/v1/gifs/search?q=${ encodeURI(category) }&limit=10&api_key=${apiKey}`;
    //     const resp = await fetch(url);
    //     const { data } = await resp.json();

    //     const gifs = data.map(img => {
    //         return {
    //             id: img.id,
    //             title: img.title,
    //             url: img.images?.downsized_medium.url
    //         }});

    //     // console.log(gifs);
    //     setImages(gifs);
    // };

    // getGifs();

    
    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>

            {/* { loading ? <p>Cargando...</p> : null } */}
            { loading && <p className="animate__animated animate__flash">Cargando...</p> }
            <div className="card-grid">
                {
                    images.map(img => (
                        <GifGridItem key={ img.id} {...img} />
                    ))
                }
            </div>
        </>
    );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}