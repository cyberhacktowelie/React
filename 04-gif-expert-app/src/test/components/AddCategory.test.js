import { AddCategory } from '../../components/AddCategory';
import { shallow } from 'enzyme';

describe('Pruebas en el componente AddCategory   AddCategory.test.js', () => {

    const setCategories = jest.fn();
    // const setCategories = () => {};
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>);

    beforeEach(() => {
        setCategories.mockClear();//jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    });


    test('Debe de mostrarse correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot();
    })

    test('Debe cambiar el valor del input', () => {
            
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        input.simulate('change', { target: { value } });
        expect( wrapper.find('input').prop('value') ).toBe(value);

    })

    test('No debe de postear la categoria si el input esta vacio', () => {
            
        const form = wrapper.find('form');
        const value = '';
        form.simulate('submit', { preventDefault() {} });
        expect( setCategories ).not.toHaveBeenCalled();

    })


    test('Debe llamar el setCategories y limpiar la caja de texto', () => {
                
            const value = 'Hola Mundo';
            const input = wrapper.find('input');
            input.simulate('change', { target: { value } });

            const form = wrapper.find('form');
            form.simulate('submit', { preventDefault(){ /* TODO document why this method 'preventDefault' is empty */ } });

            expect( setCategories ).toHaveBeenCalled();
            expect( setCategories ).toHaveBeenCalledWith(expect.any(Function));
            expect( setCategories ).toHaveBeenCalledTimes(1);
            expect( input.prop('value') ).toBe('');
    
        })

});