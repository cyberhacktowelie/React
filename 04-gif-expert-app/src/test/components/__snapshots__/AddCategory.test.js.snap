// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente AddCategory   AddCategory.test.js Debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
